perfkitbenchmarker/publisher.py

```diff
@@ -528,17 +528,18 @@ def AddSamples(self, samples, benchmark, benchmark_spec):
     """Adds data samples to the publisher.
 
     Args:
-      samples: Either a list of Sample objects (preferred) or a list of 3 or
-        4-tuples (deprecated). The tuples contain the metric name (string), the
+      samples: Either a list of Sample objects (preferred) or a list of 3, 4, or
+        5-tuples (deprecated). The tuples contain the metric name (string), the
         value (float), and unit (string) of each sample. If a 4th element is
-        included, it is a dictionary of metadata associated with the sample.
+        included, it is a dictionary of metadata associated with the sample. If
+        a 5th element is included, it is the sample's timestamp.
       benchmark: string. The name of the benchmark.
       benchmark_spec: BenchmarkSpec. Benchmark specification.
     """
     for s in samples:
       # Convert input in deprecated format to Sample objects.
       if isinstance(s, (list, tuple)):
-        if len(s) not in (3, 4):
+        if len(s) not in (3, 4, 5):
           raise ValueError(
               'Invalid sample "{0}": should be 3- or 4-tuple.'.format(s))
```

Comment:
I did this because as long as we have the legacy code in here, I felt like I should keep it up-to-date. I will update this PR to fix the message.

However, I've also been looking for users of the old code, and I believe there are none left. Once this lands, I will send a PR to remove it.